File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/common/DimensionConverter.java
package com.adadapted.android.sdk.core.common;

public class DimensionConverter {
    private final float scale;

    public DimensionConverter(final float scale) {
        this.scale = scale;
    }

    public int convertDpToPx(final int dpValue) {
        if(dpValue > 0) {
            return (int) (dpValue * scale + 0.5f);
        }

        return dpValue;
    }
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/common/DimensionConverter.kt
package com.adadapted.android.sdk.core.common

class DimensionConverter(private val scale: Float) {
    fun convertDpToPx(dpValue: Int): Int {
        return if (dpValue > 0) {
            (dpValue * scale + 0.5f).toInt()
        } else dpValue
    }

    companion object {
        private lateinit var instance: DimensionConverter

        fun getInstance(): DimensionConverter {
            return if(this::instance.isInitialized) {
                instance
            } else {
                createInstance(0f)
                instance
            }
        }

        fun createInstance(scale: Float) {
            instance = DimensionConverter(scale)
        }
    }
}